pkg/controller/controller_clusterservicebroker.go

```diff
@@ -119,13 +124,45 @@ func (c *controller) reconcileClusterServiceBrokerKey(key string) error {
 	return c.reconcileClusterServiceBroker(broker)
 }
 
+func (c *controller) updateClusterServiceBrokerClient(broker *v1beta1.ClusterServiceBroker) (osb.Client, error) {
```

Comment:
I'm thinking about a change:
 return only one value here - the error
 the brokerClientManager UpdateBrokerClient also return only error

it seems to be better, but I'd like to wait for tests and comments about the main concept - caching OSB clients